diff -urN viewcvs-oryg/lib/config.py viewcvs/lib/config.py
--- viewcvs-oryg/lib/config.py	2004-09-25 02:00:27.000000000 +0200
+++ viewcvs/lib/config.py	2004-12-17 18:24:11.080243720 +0100
@@ -213,7 +213,7 @@
     self.options.allow_tar = 0
     self.options.use_cvsgraph = 0
     self.options.cvsgraph_path = ''
-    self.options.cvsgraph_conf = os.path.join(r"<VIEWCVS_INSTALL_DIRECTORY>", "cvsgraph.conf")
+    self.options.cvsgraph_conf = "/etc/cvsgraph.conf"
     self.options.use_re_search = 0
     self.options.use_pagesize = 0
     self.options.use_localtime = 0
diff -urN viewcvs-oryg/lib/cvsdb.py viewcvs/lib/cvsdb.py
--- viewcvs-oryg/lib/cvsdb.py	2004-10-16 04:40:10.000000000 +0200
+++ viewcvs/lib/cvsdb.py	2004-12-17 18:24:11.089242352 +0100
@@ -38,7 +38,7 @@
 
 ## load configuration file, the data is used globally here
 if CONF_PATHNAME:
-  _cfg_pathname = CONF_PATHNAME
+  _cfg_pathname = "/etc/viewcvs.conf"
 else:
   # developer assistance: running from a CVS working copy
   _cfg_pathname = os.path.join(os.path.dirname(__file__), os.pardir,
diff -urN viewcvs-oryg/lib/query.py viewcvs/lib/query.py
--- viewcvs-oryg/lib/query.py	2004-10-16 02:47:41.000000000 +0200
+++ viewcvs/lib/query.py	2004-12-17 18:27:23.259028104 +0100
@@ -424,7 +424,7 @@
       data['hours'] = 2
 
     template = ezt.Template()
-    template.parse_file(os.path.join(viewcvs.g_install_dir,
+    template.parse_file(os.path.join("/usr/share/viewcvs",
                                      cfg.templates.query))
 
     server.header()
diff -urN viewcvs-oryg/lib/viewcvs.py viewcvs/lib/viewcvs.py
--- viewcvs-oryg/lib/viewcvs.py	2004-12-10 03:09:46.000000000 +0100
+++ viewcvs/lib/viewcvs.py	2004-12-17 18:28:41.070199008 +0100
@@ -107,12 +107,7 @@
 # special characters that don't need to be URL encoded
 _URL_SAFE_CHARS = "/*~"
 
-if CONF_PATHNAME:
-  # installed
-  g_install_dir = os.path.dirname(CONF_PATHNAME)
-else:
-  # development directories
-  g_install_dir = os.path.join(os.pardir, os.pardir) # typically, "../.."
+g_install_dir = os.path.dirname("/usr")
 
 
 class Request:
@@ -691,7 +686,7 @@
     tname = string.replace(tname, '%lang%', 'en')
 
   debug.t_start('ezt-parse')
-  template = ezt.Template(os.path.join(g_install_dir, tname))
+  template = ezt.Template(os.path.join("/usr/share/viewcvs", tname))
   debug.t_end('ezt-parse')
 
   template.generate(sys.stdout, data)
@@ -2103,7 +2098,7 @@
   Using this avoids the need for modifying the setup of the web server.
   """
   help_page = request.where
-  doc_directory = os.path.join(g_install_dir, "templates", "docroot")
+  doc_directory = os.path.join("/usr/share/viewcvs", "templates", "docroot")
   filename = os.path.join(doc_directory, help_page)
 
   try:
@@ -3191,7 +3186,7 @@
     cfg.set_defaults()
 
     # load in configuration information from the config file
-    pathname = CONF_PATHNAME or os.path.join(g_install_dir, 'viewcvs.conf')
+    pathname = '/etc/viewcvs.conf'
     if sapi.server:
       cfg.load_config(pathname, sapi.server.getenv('HTTP_HOST'))
     else:
diff -urN viewcvs-oryg/viewcvs.conf.dist viewcvs/viewcvs.conf.dist
--- viewcvs-oryg/viewcvs.conf.dist	2004-10-26 04:09:09.000000000 +0200
+++ viewcvs/viewcvs.conf.dist	2004-12-17 18:24:11.123237184 +0100
@@ -68,6 +68,7 @@
 # names to them. Each root should be given by a "name: path" value. Multiple
 # roots should be separated by commas and can be placed on separate lines.
 #
+# cvs_roots = cvs: /srv/cvs
 cvs_roots = cvs: /home/cvsroot
 
 #
@@ -76,6 +77,7 @@
 # by a "name: path" value. Multiple roots should be separated by
 # commas and can be placed on separate lines.
 #
+# svn_roots = svn: /srv/svn
 #svn_roots = svn: /home/svnrepos
 
 # The 'root_parents' setting specifies a list of directories in which
diff -urN viewcvs-oryg/viewcvs-install viewcvs/viewcvs-install
--- viewcvs-oryg/viewcvs-install	2004-09-28 18:30:22.000000000 +0200
+++ viewcvs/viewcvs-install	2004-12-17 18:25:11.693029176 +0100
@@ -30,7 +30,8 @@
 import StringIO
 
 # get access to our library modules
-sys.path.insert(0, os.path.join(os.path.dirname(sys.argv[0]), 'lib'))
+sys.path.append(os.path.join(os.path.dirname(sys.argv[0]), 'lib'))
+LIBRARY_DIR = sys.argv[1]
 
 import compat
 import viewcvs 
@@ -65,22 +66,22 @@
 ##       
 
 FILE_INFO_LIST = [
-    ("www/cgi/viewcvs.cgi", "www/cgi/viewcvs.cgi", 0755, 1, 0, 0),
-    ("www/cgi/query.cgi", "www/cgi/query.cgi", 0755, 1, 0, 0),
-    ("www/mod_python/viewcvs.py", "www/mod_python/viewcvs.py", 0755, 1, 0, 0),
-    ("www/mod_python/query.py", "www/mod_python/query.py", 0755, 1, 0, 0),
-    ("www/mod_python/.htaccess", "www/mod_python/.htaccess", 0755, 1, 0, 0),
-    ("standalone.py", "standalone.py", 0755, 1, 0, 0),
-    ("viewcvs.conf.dist", "viewcvs.conf", 0644, 1,
+    ("www/cgi/viewcvs.cgi", "home/services/httpd/cgi-bin/viewcvs.cgi", 0755, 1, 0, 0),
+    ("www/cgi/query.cgi", "home/services/httpd/cgi-bin/query.cgi", 0755, 1, 0, 0),
+    ("www/mod_python/viewcvs.py", "home/services/httpd/python/viewcvs.py", 0755, 1, 0, 0),
+    ("www/mod_python/query.py", "home/services/httpd/python/query.py", 0755, 1, 0, 0),
+    ("www/mod_python/.htaccess", "home/services/httpd/python/.htaccess", 0755, 1, 0, 0),
+    ("standalone.py", "usr/bin/standalone.py", 0755, 1, 0, 0),
+    ("viewcvs.conf.dist", "etc/viewcvs.conf", 0644, 1,
 	     """Note: If you are upgrading from viewcvs-0.7 or earlier: 
 The section [text] has been removed from viewcvs.conf.  The functionality
 went into the new files in subdirectory templates.""", 0),
-    ("cvsgraph.conf.dist", "cvsgraph.conf", 0644, 0, 1, 0),
+    ("cvsgraph.conf.dist", "etc/cvsgraph.conf", 0644, 0, 1, 0),
 
-    ("tools/loginfo-handler", "loginfo-handler", 0755, 1, 0, 0),
-    ("tools/cvsdbadmin", "cvsdbadmin", 0755, 1, 0, 0),
-    ("tools/svndbadmin", "svndbadmin", 0755, 1, 0, 0),
-    ("tools/make-database", "make-database", 0755, 1, 0, 0),
+    ("tools/loginfo-handler", "usr/bin/loginfo-handler", 0755, 1, 0, 0),
+    ("tools/cvsdbadmin", "usr/bin/cvsdbadmin", 0755, 1, 0, 0),
+    ("tools/svndbadmin", "usr/bin/svndbadmin", 0755, 1, 0, 0),
+    ("tools/make-database", "usr/bin/make-database", 0755, 1, 0, 0),
     ]
 
 if sys.platform == "win32":
@@ -90,8 +91,8 @@
   ])
 
 TREE_LIST = [
-  ("lib",       "lib",       0),
-  ("templates", "templates", 1),
+  ("lib",       LIBRARY_DIR + "/",             0),
+  ("templates", "usr/share/viewcvs/templates", 1),
   ]
 
 # used to escape substitution strings passed to re.sub(). re.escape() is no
diff -urN viewcvs-oryg/www/cgi/query.cgi viewcvs/www/cgi/query.cgi
--- viewcvs-oryg/www/cgi/query.cgi	2003-10-27 18:28:30.000000000 +0100
+++ viewcvs/www/cgi/query.cgi	2004-12-17 18:24:11.356201768 +0100
@@ -40,10 +40,7 @@
 
 import sys
 
-if LIBRARY_DIR:
-  sys.path.insert(0, LIBRARY_DIR)
-else:
-  sys.path[:0] = ['../../lib']	# any other places to look?
+LIBRARY_DIR = sys.argv[1]; sys.path.append(LIBRARY_DIR)
 
 #########################################################################
 
diff -urN viewcvs-oryg/www/cgi/viewcvs.cgi viewcvs/www/cgi/viewcvs.cgi
--- viewcvs-oryg/www/cgi/viewcvs.cgi	2003-09-19 00:18:28.000000000 +0200
+++ viewcvs/www/cgi/viewcvs.cgi	2004-12-17 18:24:11.361201008 +0100
@@ -40,10 +40,7 @@
 
 import sys
 
-if LIBRARY_DIR:
-  sys.path.insert(0, LIBRARY_DIR)
-else:
-  sys.path[:0] = ['../../lib']	# any other places to look?
+LIBRARY_DIR = sys.argv[1]; sys.path.append(LIBRARY_DIR)
 
 #########################################################################
 
