diff -Nur viewcvs-0.9.2.org/lib/viewcvs.py viewcvs-0.9.2/lib/viewcvs.py
--- viewcvs-0.9.2.org/lib/viewcvs.py	2004-03-03 13:52:58.000000000 +0000
+++ viewcvs-0.9.2/lib/viewcvs.py	2004-03-03 13:55:27.000000000 +0000
@@ -473,7 +473,7 @@
 def markup_stream_enscript(lang, fp):
   sys.stdout.flush()
   enscript = popen.pipe_cmds([(os.path.normpath(os.path.join(cfg.options.enscript_path,'enscript')),
-                               '--color', '-W', 'html', '-E' + lang, '-o',
+                               '--color', '--language', 'html', '-E' + lang, '-o',
                                '-', '-'),
                               ('sed', '-n', '/^<PRE>$/,/<\\/PRE>$/p')])
 
@@ -489,7 +489,7 @@
   except IOError, v:
     print "<h3>Failure during use of an external program:</h3>"
     print "<pre>"
-    print os.path.normpath(os.path.join(cfg.options.enscript_path,'enscript')) + " --color -W html -E"+lang+" -o - -"
+    print os.path.normpath(os.path.join(cfg.options.enscript_path,'enscript')) + " --color --language html -E"+lang+" -o - -"
     print "</pre>"
     raise
 
