diff -uNr viewcvs-0.9.2/lib/config.py viewcvs-0.9.2.new/lib/config.py
--- viewcvs-0.9.2/lib/config.py	Sun Feb  9 21:06:20 2003
+++ viewcvs-0.9.2.new/lib/config.py	Sun Feb  9 21:06:08 2003
@@ -154,6 +154,7 @@
     self.general.forbidden = ()
     self.general.kv_files = [ ]
     self.general.languages = ['en-us']
+    self.general.limit = 1000
 
     self.templates.directory = 'templates/directory.ezt'
     self.templates.log = 'templates/log.ezt'
diff -uNr viewcvs-0.9.2/lib/viewcvs.py viewcvs-0.9.2.new/lib/viewcvs.py
--- viewcvs-0.9.2/lib/viewcvs.py	Sun Feb  9 21:06:20 2003
+++ viewcvs-0.9.2.new/lib/viewcvs.py	Sun Feb  9 21:08:23 2003
@@ -1255,9 +1255,17 @@
   ### display a row for ".." ?
 
   rows = data['rows'] = [ ]
+  limit = cfg.general.limit
+  if query_dict.get('start'):
+    start = int(query_dict.get('start'))
+  else:
+    start = 0
+  counter = 0
 
   for file, pathname, isdir in file_data:
 
+    counter = counter + 1
+
     row = _item(href=None, graph_href=None,
                 author=None, log=None, log_file=None, log_rev=None,
                 show_log=None, state=None)
@@ -1276,7 +1284,9 @@
       row.name = file + slash
       row.type = 'unreadable'
 
-      rows.append(row)
+      if counter <= start + limit:
+        if counter > start:
+	  rows.append(row)
 
       unreadable = 1
       continue
@@ -1320,7 +1330,9 @@
       else:
         row.cvs = 'none'
 
-      rows.append(row)
+      if counter <= start + limit:
+        if counter > start:
+	  rows.append(row)
 
     else:
       # remove the ",v"
@@ -1333,7 +1345,10 @@
       info = fileinfo.get(file)
       if info == _FILE_HAD_ERROR:
         row.cvs = 'error'
-        rows.append(row)
+
+        if counter <= start + limit:
+          if counter > start:
+            rows.append(row)
 
         num_displayed = num_displayed + 1
         unreadable = 1
@@ -1356,7 +1371,6 @@
       row.rev = info[0]
       row.author = info[3]
       row.state = info[5]
-
       row.rev_href = file_url + '?rev=' + row.rev + request.amp_query
 
       row.time = html_time(request, info[1])
@@ -1368,7 +1382,9 @@
         row.show_log = 'yes'
         row.log = format_log(info[2])
 
-      rows.append(row)
+      if counter <= start + limit:
+        if counter > start:
+          rows.append(row)
 
   ### we need to fix the template w.r.t num_files. it usually is not a
   ### correct (original) count of the files available for selecting
@@ -1414,6 +1430,19 @@
     url = tar_basename + '.tar.gz?tarball=1' + request.amp_query
     data['tarball_href'] = url
 
+  data['pager'] = pager = [ ]
+  if counter > limit:
+    for i in range((counter / limit) + 1):
+      pager.append(_item(href=request.url+"/?start="+str(i*limit), page=i))
+    if start/limit > 0:
+      data['prevhref']=request.url+"/?start="+str((start/limit-1)*limit)
+    else:
+      data['prevhref']=request.url+"/?start=0"
+    if (start/limit+1)*limit <= counter:
+      data['nexthref']=request.url+"/?start="+str((start/limit+1)*limit)
+    else:
+      data['nexthref']=request.url+"/?start="+str((counter/limit)*limit)
+
   http_header()
   generate_page(request, cfg.templates.directory, data)
 
diff -uNr viewcvs-0.9.2/templates/directory.ezt viewcvs-0.9.2.new/templates/directory.ezt
--- viewcvs-0.9.2/templates/directory.ezt	Sun Dec 23 05:13:35 2001
+++ viewcvs-0.9.2.new/templates/directory.ezt	Sun Feb  9 21:06:08 2003
@@ -302,4 +302,16 @@
   <a href="[tarball_href]">Download tarball</a>
 [end]
 
+[if-any pager]
+<table width="100%" border=0 cellspacing=0 cellpadding=0>
+  <tr>
+[if-any prevhref]<td><a href="[prevhref]">&lt;&lt;</a></td>[end]
+[for pager]
+      <td><a href="[pager.href]">[pager.page]</a></td>
+[end]
+[if-any nexthref]<td><a href="[nexthref]">&gt;&gt;</a></td>[end]
+  </tr>
+</table>
+[end]
+
 [include "footer.ezt"]
